### PR TITLE
[Deps]: bump 3.16 and Node 10 minimum

### PR DESCRIPTION
**Pull request by snewcomer**
ember-data relies on external partner tests for its CI suite.  The result of a new change to convert our Model class to native classes also requires the external partner tests to follow suite.  

https://github.com/emberjs/data/pull/7226

### CODE REVIEW

**Comment by ryanto** [comment]
Awesome, thanks for doing this work!

Just a few questions.

1. Looks like some tests are failing, anyway to get those passing?

2. I noticed you removed old try scenarios, does this mean that this addon will now only support 3.12+? Should be fine if so, just want to understand.

Excited to get this merged!

**Comment by snewcomer** [comment]
Happy to help!

1. Awaiting a minor release for this library - https://github.com/ember-learn/ember-cli-addon-docs/pull/552

2. Once I get the bump in addon-docs, I'll keep adding back until it fails!

File: .eslintrc.js [hunk 1]
```diff
@@ -38,16 +49,15 @@ module.exports = {
...
+        sourceType: 'script'
       },
       env: {
         browser: false,
         node: true
       },
       plugins: ['node'],
       rules: Object.assign({}, require('eslint-plugin-node').configs.recommended.rules, {
-        // add your custom rules and overrides for node files here
+        'node/no-unpublished-require': 'off'
```
Comment (snewcomer) [inline .eslintrc.js:60]: I've looked up this issue and still not sure why it is warning.  We have `glob`, `express`, etc in devDeps 🤔 

https://github.com/mysticatea/eslint-plugin-node/issues/47

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -7,17 +7,13 @@ jobs:
       matrix:
         ember-version:
           - 'ember-default'
-          - 'ember-lts-2.16'
-          - 'ember-lts-2.18'
-          - 'ember-3.1'
-          - 'ember-lts-3.4'
-          - 'ember-lts-3.8'
           - 'ember-lts-3.12'
```
Comment (snewcomer) [inline .github/workflows/main.yml:10]: 3.10 is the minimum for classes and decorators and 3.6 is the minimum for classes.  We could install something to get us back to 3.8 but I'm not sure that is necessary if we release a new major bump

File: .travis.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ language: node_js
 sudo: false
 dist: trusty
 node_js:
-  - "8"
+  - "10"
```
Comment (snewcomer) [inline .travis.yml:6]: This will require a major bump

File: tests/integration/changing-data-render-test.js [hunk 4]
```diff
@@ -32,11 +31,8 @@ module('Integration | Changing data render test', function(hooks) {
     let serverPost = this.server.create('post', { title: 'Lorem' });
     let postId = serverPost.id;
 
-    await run(() => {
```
Comment (snewcomer) [inline tests/integration/changing-data-render-test.js:35]: `run` isn't necessary (can't remember when though)

**Review by snewcomer — COMMENTED** [review]
@ryanto This is ready for your review!  Lmk if you have any questions.

**Comment by snewcomer** [comment]
@ryanto 👋  How is this looking to you?

**Comment by snewcomer** [comment]
Hi @ryanto!  Do you happen to have time to give this a look?  We can further iterate on ember-data work once we can get this PR merged.  If not, we can temporarily patch our test suite.

**Pull request merged**